Added routing

Adamik10 opened the pull request:
- new dependency - react-router-dom
- added new static pages: about, contact, 404

Adamik10 committed:
Added routing
- new dependency - react-router-dom
- added new static pages: about, contact, 404

spaceo commented on app/src/App.tsx:
```diff
@@ -7,10 +7,12 @@ const queryClient = new QueryClient();
-    <QueryClientProvider client={queryClient}>
-      <Dashboard />
-    </QueryClientProvider>
+    <>
```
Why are the fragments used here? Just curious :)

Adamik10 replied:
I added another component/div there, so I made it to a fragment, then I removed the div, but didn't remove the fragment. Reverting it now ;)

spaceo commented on app/src/components/About.tsx:
```diff
@@ -0,0 +1,12 @@
+import React from "react";
```
I think it could make sense to make a new folder at the same level as components called pages?
And maybe to make it even more obvious we could also call the file AboutPage.tsx?

Adamik10 replied:
That sounds like a great way of doing it! On it.

spaceo commented on app/src/components/CardGrid.tsx:
```diff
@@ -33,9 +33,7 @@ const CardGrid: React.FC<Props> = ({ status, isFetching, data }) => {
-              color={`${
-                animal.animal === "crocodile" ? "#7bb853" : "#f0cc55"
-              }`}
+              color={`${animal.animal === "crocodile" ? "#7bb853" : "#f0cc55"}`}
```
I think we should put this in a map somewhere. We could make a conf/animals.ts object like:
```
export animals {
  "Crocodile": {
    color: "#....",
  },
  "Cat": {
     ...
  }
}

spaceo commented on app/src/components/Contact.tsx:
```diff
@@ -0,0 +1,12 @@
+import React from "react";
```
See my About comment :)

spaceo commented on app/src/components/Oopsie.tsx:
```diff
@@ -0,0 +1,13 @@
+import React from "react";
```
Well, you know :)

spaceo reviewed (commented):
Router for the win!
I have a few comments.
Mostly suggestions.

Adamik10 committed:
Pull request review changes
- removed unnecessary imports
- restructured the project:
  - pages folder with main pages
  - conf folder with configuration files and interfaces
  - dynamic color and type for animal icons in CardGrid.tsx

spaceo reviewed (approved):
There is still one thing that can be retrieved in the configuration.

spaceo commented on package.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "react-router-dom": "6"
```
Wrong file. It should be added to app/package.json

spaceo commented on app/src/components/CardGrid.tsx:
```diff
@@ -32,10 +33,10 @@ const CardGrid: React.FC<Props> = ({ status, isFetching, data }) => {
+              name={
+                animalIcons[animal.animal as keyof AnimalIconsInterface].name
+              }
+              color={`${animal.animal === "crocodile" ? "#7bb853" : "#f0cc55"}`}
```
I have made a suggestion to cleanup this part of the code:
https://github.com/reload/pet-dream-store/pull/9

spaceo reviewed (changes requested):
The package.json problem needs to be solved as a minimum.

Adamik10 committed:
Corrected the files according to the pull request review changes
- moved react router dependency into the app folder
- finished implementing an unfinished feature in CardGrid.tsx - Animal icon color property

Adamik10 committed:
Polished CardGrid Animal icon properties

spaceo commented:
LGTM 👍
